### ISSUE TITLE
Broken dependency - gensim

### ISSUE THREAD

**Issue by vwbusguy** [UCSB-PSTAT/ling-research-container-image#6]
The image builds but tests fail on importing gensim with a scipy incompatibility:

```
Executing the command: python -c import numpy;import gensim;import sklearn; sklearn.show_versions();import pytest;from prettytable import PrettyTable;import requests;import matplotlib;import nltk;import pandas;import arpa;import morfessor;import spacy; import torch; import keras; import tensorflow; import tgt; import textgrid;
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/opt/conda/lib/python3.12/site-packages/gensim/__init__.py", line 11, in <module>
    from gensim import parsing, corpora, matutils, interfaces, models, similarities, utils  # noqa:F401
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/conda/lib/python3.12/site-packages/gensim/corpora/__init__.py", line 6, in <module>
    from .indexedcorpus import IndexedCorpus  # noqa:F401 must appear before the other classes
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/conda/lib/python3.12/site-packages/gensim/corpora/indexedcorpus.py", line 14, in <module>
    from gensim import interfaces, utils
  File "/opt/conda/lib/python3.12/site-packages/gensim/interfaces.py", line 19, in <module>
    from gensim import utils, matutils
  File "/opt/conda/lib/python3.12/site-packages/gensim/matutils.py", line 20, in <module>
    from scipy.linalg import get_blas_funcs, triu
ImportError: cannot import name 'triu' from 'scipy.linalg' (/opt/conda/lib/python3.12/site-packages/scipy/linalg/__init__.py)
```

**Comment by ParadoxGuitarist** [comment]
gensim patched this with 4.3.3 and now requires numpy for  `triu` function. Installing via pip gets you the correct version.

**Comment by ParadoxGuitarist** [comment]
https://github.com/piskvorky/gensim/pull/3524/files
